### PR TITLE
feat: Added `enable` & `disable` Analytics APIs

### PR DESCRIPTION
**Pull request by jimblanc**
#### Description of changes

Adds the `enable` & `disable` category-level Analytics APIs.

#### Description of how you validated changes
Local testing.

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/analytics/__tests__/apis/disable.test.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+	beforeEach(() => {
+		mockEnableAnalytics.mockReset();
+		mockDisableAnalytics.mockReset();
+	});
+
+	it('should disable Analytics', () => {
+		disable();
+
+		expect(mockEnableAnalytics).not.toBeCalled();
```
Comment (cshfang) [inline packages/analytics/__tests__/apis/disable.test.ts:21]: Nit: I think you should just remove everything in this suite that has to do with `enable`. It's completely irrelevant to this api outside of our own preconceptions.

Inversely ditto for the `enable` suite

File: packages/analytics/src/utils/status.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+let AnalyticsEnabled = true;
```
Comment (cshfang) [inline packages/analytics/src/utils/status.ts:4]:
```suggestion
let analyticsEnabled = true;
```

File: packages/analytics/src/utils/status.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+export const enableAnalytics = () => (AnalyticsEnabled = true);
+
+export const disableAnalytics = () => (AnalyticsEnabled = false);
```
Comment (cshfang) [inline packages/analytics/src/utils/status.ts:8]: Nit: I think these would currently both return `true` (via returning the assignment)
```suggestion
export const enableAnalytics = () => { AnalyticsEnabled = true };

export const disableAnalytics = () => { AnalyticsEnabled = false };
```

File: packages/analytics/src/utils/status.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (cshfang) [inline packages/analytics/src/utils/status.ts:1]: Nit: `status.ts` -> `statusHelpers.ts`

File: packages/analytics/src/providers/pinpoint/apis/record.ts [hunk 5]
```diff
@@ -47,6 +50,11 @@ const logger = new Logger('Analytics');
 export const record = (input: RecordInput): void => {
 	const { appId, region } = resolveConfig();
 
+	if (!isAnalyticsEnabled()) {
+		logger.debug('Analytics is disabled, event dropped.');
```
Comment (cshfang) [inline packages/analytics/src/providers/pinpoint/apis/record.ts:54]: Nit: "Dropped" I think has some weird implications - maybe be more imperative with the wording? E.g. "event will not be recorded" or something?

File: packages/analytics/src/apis/disable.ts [hunk 6]
```diff
@@ -0,0 +1,13 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { disableAnalytics } from '../utils';
+
+/**
+ * Disables the Analytics category.
+ *
+ * @note
+ * When Analytics is disabled events will not be buffered or transmitted to your selected service. Any auto tracking
```
Comment (cshfang) [inline packages/analytics/src/apis/disable.ts:10]: Nit:
```suggestion
 * When Analytics is disabled events will not be buffered or transmitted to your selected service. Any auto-tracking
```

File: packages/analytics/__tests__/utils/status.test.ts [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+} from '../../src/utils';
+
+describe('Analytics Category Util: status utils', () => {
+	it('should indicate that Analytics is enabled by default', () => {
+		const status = isAnalyticsEnabled();
+
+		expect(status).toBeTruthy();
+	});
+
+	it('correctly toggle the Analytics status', () => {
```
Comment (cshfang) [inline packages/analytics/__tests__/utils/status.test.ts:17]: Nit:
```suggestion
	it('correctly toggles the Analytics status', () => {
```

File: packages/analytics/__tests__/utils/status.test.ts [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+	enableAnalytics,
+	disableAnalytics,
+	isAnalyticsEnabled,
+} from '../../src/utils';
+
+describe('Analytics Category Util: status utils', () => {
+	it('should indicate that Analytics is enabled by default', () => {
+		const status = isAnalyticsEnabled();
+
+		expect(status).toBeTruthy();
```
Comment (cshfang) [inline packages/analytics/__tests__/utils/status.test.ts:14]:
```suggestion
		expect(status).toBe(true);
```

File: packages/analytics/__tests__/providers/pinpoint/apis/record.test.ts [hunk 9]
```diff
@@ -16,20 +17,31 @@ import {
 	config,
 } from './testUtils/data';
 
+jest.mock('../../../../src/utils', () => {
+	const original = jest.requireActual('../../../../src/utils'); // Step 2.
```
Comment (cshfang) [inline packages/analytics/__tests__/providers/pinpoint/apis/record.test.ts:21]: Why was this necessary? 🤔  We should try to avoid requiring the actual implementation if we can avoid it

Comment (jimblanc) [reply packages/analytics/__tests__/providers/pinpoint/apis/record.test.ts:21]: Yeah it was needed for user agent string generation

Comment (cshfang) [reply packages/analytics/__tests__/providers/pinpoint/apis/record.test.ts:21]: Imo we should try to just mock that out instead of requiring actual if possible. But not a blocker

File: packages/analytics/__tests__/utils/status.test.ts [hunk 10]
```diff
@@ -0,0 +1,31 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
```
Comment (cshfang) [inline packages/analytics/__tests__/utils/status.test.ts:3]: Nit: Align filename with file under test

**Review by cshfang — APPROVED** [review]

File: packages/analytics/src/apis/enable.ts [hunk 11]
```diff
@@ -0,0 +1,12 @@
...
+
+import { enableAnalytics } from '../utils';
+
+/**
+ * Enables the Analytics category to permit the transmission of events.
+ *
+ * @note
+ * Analytics is enabled by default. You do not need to call this API unless you have disabled Analytics.
+ */
+export const enable = () => enableAnalytics();
```
Comment (HuiSF) [inline packages/analytics/src/apis/enable.ts:12]: nit: saving some code bytes? 😅

```suggestion
export const enable = enableAnalytics;
```

Comment (jimblanc) [reply packages/analytics/src/apis/enable.ts:12]: my man

**Review by HuiSF — APPROVED** [review]

File: packages/analytics/__tests__/providers/pinpoint/apis/record.test.ts [hunk 12]
```diff
@@ -16,20 +20,28 @@ import {
 	config,
 } from './testUtils/data';
 
+jest.mock('../../../../src/utils');
```
Comment (cshfang) [inline packages/analytics/__tests__/providers/pinpoint/apis/record.test.ts:23]: 🤩

**Review by cshfang — APPROVED** [review]
